File: JavaScript projects/JavaScript Basics/05.While Loops/lab/08_graduation.js
function graduation(input){
    let sumFails = 0;
    let nameStudent = input[0];
    let averageGrade = 0;
    let yearlyGrade = Number(input[1]);
    let index = 1;
    let gradeYear = 1; 

    while ((gradeYear < 13) && sumFails < 2){
        yearlyGrade = Number(input[index]);

        if (yearlyGrade < 4.00){
            sumFails ++;
            index ++;
            continue;
        }
        averageGrade += yearlyGrade;
        gradeYear ++;
        index++;
        
    }
        if (sumFails === 2){
        console.log(`${nameStudent} has been excluded at ${gradeYear} grade`);
    }else{
        console.log(`${nameStudent} graduated. Average grade: ${(averageGrade / 12).toFixed(2)}`)
    }
}
//graduation(["Gosho","5","5.5","6","5.43", "5.5", "6", "5.55", "5", "6", "6", "5.43", "5"]);
graduation(["Mimi", "5", "6", "5","6", "5", "6", "6", "2", "3"])

File: Python projects/Python Basics/05_while_loops/lab/08_graduation.py
name = str(input())

sum_grades = 0.0
year = 1
failed_attempts = 0
failed = False

while year < 13:
    school_grade = float(input())
    if school_grade < 4.0 and failed_attempts == 1:
        failed = True
        break
    elif school_grade < 4.0:
        failed_attempts += 1
        continue
    sum_grades += school_grade
    year += 1

if failed:
    print(f"{name} has been excluded at {year} grade")
else:
    print(f"{name} graduated. Average grade: {(sum_grades / (year - 1)):.2f} ")
